src/data/polynomial/degree/basic.lean

```diff
@@ -170,6 +170,9 @@ by rw [← single_eq_C_mul_X, degree, monomial, support_single_ne_zero ha]; refl
 lemma degree_monomial_le (n : ℕ) (a : R) : degree (C a * X ^ n) ≤ n :=
 if h : a = 0 then by rw [h, C_0, zero_mul]; exact bot_le else le_of_eq (degree_monomial n h)
 
+@[simp] lemma nat_degree_monomial {a : R} (n : ℕ) (ha : a ≠ 0) : nat_degree (C a * X ^ n) = n :=
```

Comment:
```suggestion
@[simp] lemma nat_degree_C_mul_X_pow {a : R} (n : ℕ) (ha : a ≠ 0) : nat_degree (C a * X ^ n) = n :=
```
The lemma above this one has a "bad" name. Since `monomial` is an explicitly defined term, we shouldn't abuse it as name for things about `C_mul_X_pow`.

Reply:
A general comment: I think this is an issue in general across the polynomial files. There are multiple lemmas which use `monomial` when referring to `C a * X^n`. Here are some of the ones that I found.

* `polynomial.derivative_monomial`
* `polynomial.degree_monomial`
* `polynomial.degree_monomial_le`
* `polynomial.leading_coeff_monomial`
* `polynomial.trailing_degree_monomial`
* `polynomial.monomial_le_trailing_degree`

Reply:
I like the word "term" for "monomial with a coefficient", but it seems that the Lean way is "C_mul_X_pow".  I agree that the Lean way is more descriptive, so I am happy to comply with the convention!  I will try to be more careful and not slip into a name that I am used to!